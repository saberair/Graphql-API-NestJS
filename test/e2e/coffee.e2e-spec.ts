import { getApolloServer } from '@nestjs/apollo';
import gql from 'graphql-tag';
import { CreateCoffeeInput } from '../../src/coffee/dto/create-coffee.input';
import { UpdateCoffeeInput } from 'src/coffee/dto/update-coffee.input';
import { AppFixture, createAppFixture } from './e2e-setup';

describe('Coffee (e2e)', () => {
  let appFixture: AppFixture;
  let apolloServer: ReturnType<typeof getApolloServer>;

  afterAll(async () => {
    appFixture.destroy();
  });

  beforeAll(async () => {
    appFixture = await createAppFixture();
    apolloServer = appFixture.apolloServer;
  });

  beforeEach(async () => {
    await appFixture.reset();
  });

  it('creates a new coffee', async () => {
    const Create_Coffee_Mutation = gql`
      mutation createCoffee($createCoffeeInput: CreateCoffeeInput!) {
        createCoffee(createCoffeeInput: $createCoffeeInput) {
          id
          name
          price
          size
        }
      }
    `;

    const createCoffeeInput: CreateCoffeeInput = {
      name: 'Express',
      price: 3,
      size: 'Big',
    };

    const response = await apolloServer.executeOperation({
      query: Create_Coffee_Mutation,
      variables: { createCoffeeInput },
    });

    expect(response).toMatchSnapshot({
      body: {
        singleResult: {
          data: {
            createCoffee: {
              id: expect.any(Number),
            },
          },
        },
      },
    });
  });

  it('updates coffee', async () => {
    const Update_Coffee_Mutation = gql`
      mutation updateCoffee($updateCoffeeInput: UpdateCoffeeInput!) {
        updateCoffee(updateCoffeeInput: $updateCoffeeInput) {
          id
          name
          price
          size
        }
      }
    `;

    const updateCoffeeInput: UpdateCoffeeInput = {
      id: 1,
      name: 'Express',
      price: 4,
      size: 'Small',
    };

    const response = await apolloServer.executeOperation({
      query: Update_Coffee_Mutation,
      variables: { updateCoffeeInput },
    });

    expect(response).toMatchSnapshot();
  });

  it('removes a coffee', async () => {
    const Remove_Coffee_Mutation = gql`
      mutation removeCoffee($id: Int!) {
        removeCoffee(id: $id) {
          id
          name
          price
          size
        }
      }
    `;

    const response = await apolloServer.executeOperation({
      query: Remove_Coffee_Mutation,
      variables: { id: 2 },
    });

    expect(response).toMatchSnapshot();
  });

  it('finds all coffees', async () => {
    const Find_All_Coffees_Query = gql`
      query FindAllCoffees {
        findAllCoffees {
          id
          name
          price
          size
        }
      }
    `;

    const response = await apolloServer.executeOperation({
      query: Find_All_Coffees_Query,
    });

    expect(response).toMatchSnapshot();
  });

  it('finds one coffee', async () => {
    const Find_One_Coffee_Query = gql`
      query FindOneCoffee($id: Int!) {
        findOneCoffee(id: $id) {
          id
          name
          price
          size
        }
      }
    `;

    const response = await apolloServer.executeOperation({
      query: Find_One_Coffee_Query,
      variables: { id: 1 },
    });

    expect(response).toMatchSnapshot();
  });
});
