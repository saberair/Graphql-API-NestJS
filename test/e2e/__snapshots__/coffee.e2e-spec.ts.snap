// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coffee (e2e) creates a new coffee 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "createCoffee": Object {
          "id": Any<Number>,
          "name": "Express",
          "price": 3,
          "size": "Big",
        },
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;

exports[`Coffee (e2e) finds all coffees 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "findAllCoffees": Array [
          Object {
            "id": 1,
            "inventoryCount": 40,
            "name": "Cappucino",
            "price": 4,
            "size": "Big",
          },
          Object {
            "id": 2,
            "inventoryCount": 30,
            "name": "Expresso",
            "price": 3,
            "size": "Small",
          },
        ],
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;

exports[`Coffee (e2e) finds one coffee 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "findOneCoffee": Object {
          "id": 1,
          "inventoryCount": 40,
          "name": "Cappucino",
          "price": 4,
          "size": "Big",
        },
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;

exports[`Coffee (e2e) removes a coffee 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "removeCoffee": Object {
          "id": 2,
          "inventoryCount": 30,
          "name": "Expresso",
          "price": 3,
          "size": "Small",
        },
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;

exports[`Coffee (e2e) updates coffee 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "updateCoffee": Object {
          "id": 1,
          "inventoryCount": 40,
          "name": "Express",
          "price": 4,
          "size": "Small",
        },
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;
